### PR TITLE
I1695 - libsodium

### PR DESCRIPTION
**Pull request by littledice**
Move libsodium docker logic out of the API and instead pull the libsodium image from the registry where needed.

### CODE REVIEW

File: src/app/build.gradle [hunk 1]
```diff
@@ -51,9 +57,9 @@ dependencies {
 }
 
 distDocker {
-    baseImage = "libsodium"
+    baseImage = base_image
     dependsOn = [build]
-    push = true
+    push = false
```
Comment (MartinEden) [inline src/app/build.gradle]: Why has this changed?

File: blackbox.Dockerfile [hunk 2]
```diff
@@ -1,4 +1,5 @@
-FROM libsodium
+ARG libsodium_image=docker.montagu.dide.ic.ac.uk:5000/libsodium:i1695
```
Comment (MartinEden) [inline blackbox.Dockerfile]: Must remember to return this to master

File: src/app/build.gradle (hunk 1 above)
Comment (littledice) [reply src/app/build.gradle]: oops, local testing

**Comment by littledice** [comment]
Like read it in from a text file? I doubt it will change very often (or ever?) so I'd propose keeping it simple as is.

**Comment by MartinEden** [comment]
Okay

**Pull request merged**